test/FunctionalTests/MvcTests/TagHelperSampleTest.cs

```diff
@@ -0,0 +1,252 @@
+                            new FormUrlEncodedContent(Enumerable.Empty<KeyValuePair<string, string>>()));
+                        await client.PostAsync(
+                            "http://localhost/Movies/UpdateCriticsQuotes",
+                            new FormUrlEncodedContent(Enumerable.Empty<KeyValuePair<string, string>>()));
+
+                        return await client.GetAsync("http://localhost/Movies/Index");
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(MoviesControllerPageData))]
+        public async Task MoviesController_Pages_ReturnSuccess(Func<HttpClient, Task<HttpResponseMessage>> requestPage)
+        {
+            // Act
+            var response = await requestPage(Client);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task TagHelperController_ConditionalComment_ReturnsExpectedContent()
+        {
+            // Arrange
+            var expectedMediaType = MediaTypeHeaderValue.Parse("text/html; charset=utf-8");
+            var outputFile = "FunctionalTests.resources.TagHelperSample.Web.TagHelper.ConditionalComment.html";
+            var expectedContent = GetResourceContent(outputFile);
+
+            // Act
+            var response = await Client.GetAsync("http://localhost/TagHelper/ConditionalComment");
+            var responseContent = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(expectedMediaType, response.Content.Headers.ContentType);
+
+            Assert.Equal(expectedContent, responseContent, ignoreLineEndingDifferences: true);
+        }
+
+        private static void AssertRedirectsToHome(HttpResponseMessage response)
+        {
+            Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
+            var redirectLocations = response.Headers.GetValues("Location");
+            var redirectsTo = Assert.Single(redirectLocations);
+            Assert.Equal("/", redirectsTo, StringComparer.Ordinal);
+        }
+
+        private static HttpContent CreateUserFormContent(
+            string name,
+            string dateOfBirth,
+            int yearsEmployeed,
+            string blurb)
+        {
+            var form = new List<KeyValuePair<string, string>>()
+            {
+                new KeyValuePair<string, string>("Name", name),
+                new KeyValuePair<string, string>("DateOfBirth", dateOfBirth),
+                new KeyValuePair<string, string>("YearsEmployeed", yearsEmployeed.ToString()),
+                new KeyValuePair<string, string>("Blurb", blurb),
+            };
+            var content = new FormUrlEncodedContent(form);
+
+            return content;
+        }
+
+        private static string GetResourceContent(string name)
```

Comment:
GENERATE_BASELINES seems a bit much for sample tests. The content shouldn't change that often